dropshot/src/server.rs

```diff
@@ -981,6 +990,33 @@ async fn http_request_handle<C: ServerContext>(
             }
         }
     };
+
+    if server.config.compression
+        && is_compressible_content_type(response.headers())
+    {
+        // Add Vary: Accept-Encoding header for all compressible content
+        // types. This needs to be there even if the response ends up not being
+        // compressed because it tells caches how to handle the possibility that
+        // responses could be compressed or not.
+        //
+        // This tells caches (like browsers and CDNs) that the response content
+        // depends on the value of the Accept-Encoding header. Without this, a
+        // cache might mistakenly serve a compressed response to a client that
+        // cannot decompress it, or serve an uncompressed response to a client
+        // that could have benefited from compression.
```

Comment:
```suggestion
        // Add Vary: Accept-Encoding header for all compressible content
        // types. This needs to be there even if the response ends up not being
        // compressed because it tells caches (like browsers and CDNs) how to handle the possibility that
        // responses could be compressed or not
        // depending on the value of the Accept-Encoding header. Without this, a
        // cache might mistakenly serve a compressed response to a client that
        // cannot decompress it, or serve an uncompressed response to a client
        // that could have benefited from compression.
```

Felt redundant. Not sure mine is better; do what you will here.